.github/workflows/run_integration_tests.yml

```diff
@@ -2,13 +2,406 @@ name: Run Integration Tests
+        required: true
+        default: 'ALL'
+      unbounded-test-namespaces:
+        description: 'A comma-separated, single-quoted list of unbounded integration test namespaces to run (e.g. "''MsSql'', ''Couchbase''").  If not specified, all unbounded tests will be run.'
+        required: true
+        default: 'ALL'
+      # iterations:
+      #   description: 'Number of times to execute each test, regardless of success or failure.  Defaults to 1.'
+      #   required: true
+      #   default: 1
+
+env:
+  DOTNET_NOLOGO: true
 
 jobs:
 
-  hello-world:
-    name: Placeholder job
+  get-external-artifacts:
+    name: Get and Publish Agent and Test Artifacts Locally
     runs-on: ubuntu-latest
     steps:
-    - name: Hello World
-      run: echo "Hello world!"
-      shell: bash
+      - name: Download Agent Home Folders
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          workflow: all_solutions.yml
+          run_id: ${{ github.event.inputs.run_id }}
+          name: homefolders
+          path: ${{ github.workspace }}
+          repo: ${{ github.repository }}
+      
+      - name: Upload Agent Home Folders Locally
+        uses: actions/upload-artifact@v2
+        with:
+          name: homefolders
+          path: ${{ github.workspace }}/newrelichome*
+          if-no-files-found: error
+
+      - name: Print input vars
+        run: |
+          echo "github.event.inputs.solutions = ${{ github.event.inputs.solutions }}"
+          echo "github.event.inputs.integration-test-namespaces = ${{ github.event.inputs.integration-test-namespaces }}"
+          echo "github.event.inputs.unbounded-test-namespaces = ${{ github.event.inputs.unbounded-test-namespaces }}"
+
+      - name: Download Integration Tests
+        if: github.event.inputs.solutions == 'Both' || github.event.inputs.solutions == 'IntegrationTests'
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          workflow: all_solutions.yml
+          run_id: ${{ github.event.inputs.run_id }}
+          name: integrationtests
+          path: ${{ github.workspace }}
+          repo: ${{ github.repository }}
+      
+      - name: Upload Integration Tests Locally
+        if: github.event.inputs.solutions == 'Both' || github.event.inputs.solutions == 'IntegrationTests'
+        uses: actions/upload-artifact@v2
+        with:
+          name: integrationtests
+          path: |
+            ${{ github.workspace }}/test.runsettings  # Force the artifacts to use repo root as root of package.
+            ${{ github.workspace }}/tests/Agent/IntegrationTests/**/bin/**/*
+            ${{ github.workspace }}/tests/Agent/IntegrationTests/**/Deploy/**/*
+            !${{ github.workspace }}/tests/Agent/IntegrationTests/**/obj/**/*
+          if-no-files-found: error
+
+      - name: Clear integration tests from workspace
```

Comment:
I don't think we need this since the runner will be destroyed at the end of the job (two steps after this one).

Reply:
The reason this step is in there is to avoid including the "bounded" integration test assets in what gets uploaded for the unbounded integration test assets in the subsequent steps.  I can put a comment in the workflow to help clarify that.

Reply:
Makes sense!